presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -667,6 +690,36 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        if (totalSegments == 1) {
+            return getPartitions(databaseName, tableName, expression, null);
+        }
+
+        // Do parallel partition fetch.
+        CompletionService<List<Partition>> completionService = new ExecutorCompletionService<>(executor);
+        IntStream.range(0, totalSegments)
+            .mapToObj(s -> new Segment().withSegmentNumber(s).withTotalSegments(totalSegments))
+            .forEach(segment -> completionService.submit(() -> getPartitions(databaseName, tableName, expression, segment)));
+
+        List<Partition> partitions = new ArrayList<>();
+        try {
+            for (int i = 0; i < totalSegments; i++) {
+                Future<List<Partition>> futurePartitions = completionService.take();
+                partitions.addAll(futurePartitions.get());
+            }
+        }
+        catch (ExecutionException | InterruptedException e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            throw new PrestoException(HIVE_METASTORE_ERROR, "Exception when getting partitions", e);
+        }
+        Collections.sort(partitions, Comparator.comparing(p -> join(",", p.getValues())));
```

Comment:
We shouldn't need to convert to string here, as `List<String>` is naturally comparable with the same semantics

Reply:
Also this should probably be `partitions.sort()` rather than `Collections.sort(partitions)`

Reply:
Will change to partitions.sort(). Not using the string would require me to write a custom comparator that compares List<String> since List<Comparable> is not Comparable. Something like:

```
partitions.sort((p1, p2) -> {
        	List<String> values1 = p1.getValues();
        	List<String> values2 = p2.getValues();
        	if (values1.size() != values2.size()) {
        		return values1.size() - values2.size();
                 }
        	for (int i = 0; i < values1.size(); i++) {
                     int c = values1.get(i).compareTo(values2.get(i));
                     if (c != 0) {
                         return c;
                      }
		}
                 return 0;
});
```

Does that sound reasonable?

Reply:
`partitions.sort(com.google.common.collect.Ordering.natural().lexicographical())`

(there should be a way to do this without guava... but i didn't find it.)